### PR TITLE
Update development to 2.1 isValidSignature with magic salt

### PR DESCRIPTION
**Pull request by dekz**
## Description

Our local development contracts (and therefor ganache snapshots and tools) use the old signatures without magic salt values.

## Checklist:

-   [x] @jalextowle you mentioned wanting to rewrite as solidity?

### CODE REVIEW

File: contracts/exchange/contracts/src/MixinSignatureValidator.sol [hunk 1]
```diff
@@ -244,7 +244,17 @@ contract MixinSignatureValidator is
             hash,
             signature
         );
+        bytes32 magicValue = bytes32(bytes4(keccak256("isValidWalletSignature(bytes32,address,bytes)")));
```
Comment (abandeali1) [inline contracts/exchange/contracts/src/MixinSignatureValidator.sol]: Can we add a comment with the `bytes4` constants for both of these signature types?

File: contracts/exchange/contracts/src/MixinSignatureValidator.sol [hunk 2]
```diff
@@ -244,7 +244,17 @@ contract MixinSignatureValidator is
...
         assembly {
+            if iszero(extcodesize(walletAddress)) {
```
Comment (abandeali1) [inline contracts/exchange/contracts/src/MixinSignatureValidator.sol]: Let's add a comment that the `extcodesize` checks are redundant, but were added as an extra safety measure.

File: packages/order-utils/src/signature_utils.ts [hunk 3]
```diff
@@ -169,12 +174,16 @@ export const signatureUtils = {
...
 
         const validatorContract = new IValidatorContract(signerAddress, provider);
-        const isValid = await validatorContract.isValidSignature.callAsync(
-            data,
-            signerAddress,
-            validatorSignature.signature,
-        );
-        return isValid;
+        try {
+            const magicValue = await validatorContract.isValidSignature.callAsync(
```
Comment (abandeali1) [inline packages/order-utils/src/signature_utils.ts]: PR #2017 fixed a bug where we weren't checking if the validator contract was approved in the `Exchange` contract. We should really just be calling `Exchange.isValidSignature` here to ensure that is being checked with a single RPC call.

Comment (dekz) [reply packages/order-utils/src/signature_utils.ts]: We should just refactor this to use devutils entirely right? Is that currently checking if the Validator is approved?

Comment (dekz) [reply packages/order-utils/src/signature_utils.ts]: Let's follow up after #2017 in #2083

**Pull request merged**